### DISCUSSION TITLE
forecasting multi step return of intra day stock price

### DISCUSSION THREAD

**Discussion by sumonsamanta123** [ourownstory/neural_prophet#281] [General]
Anyone tried neural prophet to forecast stock return from intra day time series(5 minutes interval)? I am not getting any good result.
Attached sample time series. I am trying to use return (price(t) - price(t-1))/price(t-1) for time up to 3:30, use this to fit neural prophet and then predict the price between 3:30 to 4. Forecast does not seem to change direction(positive for all vs positive/negative in the actual return). Just using ARMA, I got better forecast(although not close to the actual return). Please note - I am not expecting to match actual return series(it's impossible), just the direction of the change is good enough. 

[intraday_5min_AMZN.txt](https://github.com/ourownstory/neural_prophet/files/6026636/intraday_5min_AMZN.txt)

**Comment by ourownstory** [comment]
Stock prices are by large not predictable by past values. 

For the sake of this experiment: What was your NP model definition?

**Reply by sumonsamanta123** [reply]
I think stock price depends on other related time series and also news feed (i.e social media twitter post, facebook, cnbc, blommberg news etc - I think the challenge is how quickly all these news feed can be analyzed and compute a overall sentiment index of the company). I am trying LSTM and see on using multi variate time series input. 

ts_df_test has observed stock price till 3:30, ts_df_forecast is between 3:30 and 4. I am trying to compare last 30 minutes price with forecast from NP. 
----------------------------------------------------
m = NeuralProphet(n_lags=5,n_changepoints=10,yearly_seasonality=False,weekly_seasonality=False)
metrics = m.fit(ts_df_test,valid_p=0.3, freq='5min') 
future = m.make_future_dataframe(ts_df_forecast)
forecast = m.predict(future)

**Reply by ourownstory** [reply]
That's also my understanding about stock data - it's driven by perceived future value, not past prices. 

If you have those related time-series, you can add them with `add_lagged_regressor`, as events or future regressors.

Also, I'd use a much larger dataset.
